libs/barista-components/chart/src/tooltip/chart-tooltip-position.ts

```diff
@@ -0,0 +1,110 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DtChart, DtChartTooltipData } from '../..';
+import { PlotBackgroundInfo } from '../utils';
+import { ConnectedPosition } from '@angular/cdk/overlay';
+import { InjectionToken } from '@angular/core';
+import { isDefined } from '@dynatrace/barista-components/core';
+
+/** InjectionToken of the Tooltip options. */
+export const DT_CHART_TOOLTIP_CONFIG = new InjectionToken<string>(
+  'DT_CHART_TOOLTIP_CONFIG',
+);
+
+export interface DtChartTooltipConfig {
+  positionFunction?: DtChartTooltipPositionFn;
+}
+
+export type DtChartTooltipPositionFn = (
+  data: DtChartTooltipData,
+  chart: DtChart,
+  plotBackgroundInfo?: PlotBackgroundInfo,
+) => { x: number; y: number };
+
+/**
+ * Calculate an origin point that can be used to position the tooltip.
+ */
+export const getDefaultTooltipPosition = (
+  data: DtChartTooltipData,
+  chart: DtChart,
+  plotBackgroundInfo: PlotBackgroundInfo,
+): { x: number; y: number } => {
+  const containerElement: HTMLElement = chart._container.nativeElement;
+  const containerElementBB = containerElement.getBoundingClientRect();
+  const { x, y } = getHighchartsTooltipPosition(data, plotBackgroundInfo);
+  return {
+    x: containerElementBB.left + x,
+    y: containerElementBB.top + y,
+  };
+};
+
+/**
+ * highcharts provides the tooltip position differently depending on the series type
+ * Pie chart: data.point.point.tooltipPos[x, y]
+ * Category: data.points[0].point.tooltipPos[x, whatever, whatever]
+ * Mixed multiple series(line, column): data.points[0].point.tooltipPos[x, whatever, whatever]
+ * Area as first: data.points[0].point.x => xAxis.toPixel(x)
+ */
+const getHighchartsTooltipPosition = (
+  data: DtChartTooltipData,
+  plotBackgroundInfo: PlotBackgroundInfo,
+): { x: number; y: number } => {
+  const isPieChart = !isDefined(data.points);
+  const hasAreaFirstSeries =
+    data.points &&
+    data.points[0].point &&
+    !(data.points[0].point as any).tooltipPos;
+  let x: number;
+  // set y position for all charts in the middle of the plotbackground vertically
+  // tslint:disable-next-line:no-magic-numbers
+  let y = plotBackgroundInfo.height / 2 + plotBackgroundInfo.top;
+  if (isPieChart) {
+    const tooltipPos = (data.point!.point as any).tooltipPos;
+    x = tooltipPos![0];
+    // override the y position for pie charts
+    y = tooltipPos![1];
+  } else if (hasAreaFirstSeries) {
+    const point = data.points![0].point;
+    const xAxis = data.points![0].series!.xAxis;
+    x = xAxis.toPixels(point.x as number, false);
+  } else {
+    x = (data.points![0].point as any).tooltipPos![0] + plotBackgroundInfo.left;
+  }
+
+  return { x, y };
+};
+
+/** Default horizontal offset for the tooltip */
+export const DT_CHART_TOOLTIP_DEFAULT_OFFSET = 10;
+
+/** Positions for the chart tooltip  */
+export const DEFAULT_DT_CHART_TOOLTIP_POSITIONS: ConnectedPosition[] = [
```

Comment:
```suggestion
export const DT_CHART_DEFAULT_TOOLTIP_POSITIONS: ConnectedPosition[] = [
```
Could you please switch this around to prefix public API with DT to fulfill our contract to our consumers.